### PR TITLE
Add extension information to snippet picker

### LINKED ISSUE #138105: insert snippet picker list show where the snippet from
Snippets may from extension or builtin, it would be helpful if display where the snippet from.

image

### PR DESCRIPTION
**Pull request by wengj9**
This PR fixes #138105.

To test this:
* Launch VSCode and open 'Insert Snippet' from the Command Palette
* Depending on installed extensions/language of the open file, you should see some snippets from installed extensions (example: 
Screen Shot 2021-12-10 at 14 13 35)

### CODE REVIEW

File: src/vs/workbench/contrib/snippets/browser/insertSnippet.ts [hunk 1]
```diff
@@ -180,6 +180,8 @@ class InsertSnippetAction extends EditorAction {
 					result.push({ type: 'separator', label });
 				}
 
+
+
```
Comment (jrieken) [inline src/vs/workbench/contrib/snippets/browser/insertSnippet.ts:184]: nit: extra new lines

**Review by jrieken — COMMENTED** [review]
Thanks. This is looking very good. Left some nit comments, just in case you want to tackle those

File: src/vs/workbench/contrib/snippets/browser/insertSnippet.ts [hunk 2]
```diff
@@ -164,14 +164,14 @@ class InsertSnippetAction extends EditorAction {
...
-				if (!prevSnippet || prevSnippet.snippetSource !== snippet.snippetSource) {
+				if (!prevSnippet || prevSnippet.snippetSource !== snippet.snippetSource || prevSnippet.source !== snippet.source) {
 					let label = '';
 					switch (snippet.snippetSource) {
 						case SnippetSource.User:
 							label = nls.localize('sep.userSnippet', "User Snippets");
 							break;
 						case SnippetSource.Extension:
-							label = nls.localize('sep.extSnippet', "Extension Snippets");
+							label = nls.localize('sep.extSnippet', "{0}", snippet.source);
```
[the remark this answers is not in the document]
Comment (wengj9) [reply src/vs/workbench/contrib/snippets/browser/insertSnippet.ts]: Resolved with latest push

File: src/vs/workbench/contrib/snippets/browser/insertSnippet.ts (hunk 1 above)
Comment (wengj9) [reply src/vs/workbench/contrib/snippets/browser/insertSnippet.ts]: Resolved with latest push

**Comment by wengj9** [comment]
I also noticed an Electron unit test was failing earlier on Actions but it wasn't on my local. Should I be concerned with that?

**Review by jrieken — APPROVED** [review]

**Pull request merged**